nntrainer/graph/graph_core.h

```diff
@@ -261,6 +264,7 @@ class GraphCore {
   std::vector<std::shared_ptr<GraphNode>> output_list;
   std::vector<std::shared_ptr<GraphNode>>
     node_list;                                    /**< Unordered Node List  */
+  std::unordered_map<std::string, int> node_map;  /**< Unordered Node map  */
```

Comment:
what's the usage difference between node_names and node_map? I wonder if the node_names is removed.

Reply:
node_map provides the location of the node in the given list of nodes for faster access. node_names provides assurance that each names in node are unique.

You are right in thinking about their overlap. I think the functionality to ensure all names are unique can be done just with node_map. I will do this in a new PR soon.

Reply:
Now as you mention it, maybe node_map and node_list can be combined as well.
In a separate PR, I will combine as many as of these structures.

Reply:
great. thanks for your consideration.